### PR TITLE
Alias `foreach` to `traverse` and `collectAll` to `sequence` (#1121)

### LINKED ISSUE #1121: Add traverse and sequence aliases
Try as I might, I keep reaching for these names instead of `foreach` and `collect`.

What do everyone think of re-adding these aliases?

A data point in favour of this is the equivalence with `Future.traverse`, `Future.sequence`.

These need to be added for:
- Exit
- ZIO
- ZManaged

### PR DESCRIPTION
**Pull request by dsebban**
Resolves #1121

### CODE REVIEW

File: core/shared/src/main/scala/zio/Exit.scala [hunk 1]
```diff
@@ -206,6 +206,12 @@ object Exit extends Serializable {
         .map(_.reverse)
     }
 
+  /**
+   *  Alias for `collectAll`.
```
Comment (iravid) [inline core/shared/src/main/scala/zio/Exit.scala]: Can you use `[[ZIO.collectAll]]` for the scaladoc references?

File: core/shared/src/main/scala/zio/Exit.scala [hunk 2]
```diff
@@ -206,6 +206,12 @@ object Exit extends Serializable {
         .map(_.reverse)
     }
 
+  /**
+   *  Alias for `[[ZIO.collectAll]]`.
```
Comment (iravid) [inline core/shared/src/main/scala/zio/Exit.scala]: Thanks! the backticks are unnecessary when you have the square brackets

Comment (dsebban) [reply core/shared/src/main/scala/zio/Exit.scala]: I generated the docs and checked the links work

Comment (iravid) [reply core/shared/src/main/scala/zio/Exit.scala]: Ah that's great then

**Pull request merged**